codex-rs/tui/src/bottom_pane/approval_modal_view.rs

```diff
@@ -59,3 +67,42 @@ impl<'a> BottomPaneView<'a> for ApprovalModalView<'a> {
         None
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::app_event::AppEvent;
+    use std::path::PathBuf;
+    use std::sync::mpsc::channel;
+
+    fn make_sender() -> AppEventSender {
+        let (tx, _rx) = channel::<AppEvent>();
+        AppEventSender::new(tx)
+    }
+
+    fn make_exec_request() -> ApprovalRequest {
+        ApprovalRequest::Exec {
+            id: "test".to_string(),
+            command: vec!["echo".to_string(), "hi".to_string()],
+            cwd: PathBuf::from("/tmp"),
+            reason: None,
+        }
+    }
+
+    #[test]
+    fn ctrl_c_aborts_and_clears_queue() {
+        let tx = make_sender();
```

Comment:
FYI, I would not use the `make_sender()` helper and just do the following:

```suggestion
        let (tx, _rx) = channel::<AppEvent>();
```

The reason is that, with `make_sender()`, `_rx` is dropped when `make_sender()` exits, which means that `tx.send()` should fail with `Err(SendError)`, which I don't think is what you want?

If you take the suggestion above, `_rx` will not be dropped until the block exits, which in this case, is the lifetime of the `ctrl_c_aborts_and_clears_queue()` test itself.